File: src/main/java/com/basic/kotlin/sec2/FunctionExample.java
package com.basic.kotlin.sec2;

public class FunctionExample {

    public void repeat(String str, int number, boolean useNewLine) {
        for (int i = 0; i < number; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }

    public void repeat(String str, int number) {
        repeat(str, number, true);
    }

    public void repeat(String str) {
        repeat(str, 3);
    }
}

File: src/main/kotlin/com/basic/kotlin/sec2/FunctionExample.kt
package com.basic.kotlin.sec2

fun max(a: Int, b: Int) = if (a > b) a else b

fun repeat(str: String, count: Int, useNewLine: Boolean = true) {
    for (i in 1..count) {
        when {
            useNewLine -> println(str)
            else -> print(str)
        }
    }
}
